src/main/java/org/apache/sling/resourceresolver/impl/mapping/VanityPathHandler.java

```diff
@@ -222,30 +223,43 @@ private String getMapEntryRedirect(final MapEntry mapEntry) {
     }
 
     boolean doRemoveVanity(final String path) {
-        final String actualContentPath = getActualContentPath(path);
-        final List <String> l = vanityTargets.remove(actualContentPath);
-        if (l != null){
-            for (final String s : l){
-                final List<MapEntry> entries = this.resolveMapsMap.get(s);
-                if (entries!= null) {
-                    for (final Iterator<MapEntry> iterator = entries.iterator(); iterator.hasNext(); ) {
-                        final MapEntry entry = iterator.next();
-                        final String redirect = getMapEntryRedirect(entry);
-                        if (redirect != null && redirect.equals(actualContentPath)) {
-                            iterator.remove();
-                        }
-                    }
+        String actualContentPath = getActualContentPath(path);
+        List<String> targets = this.vanityTargets.remove(actualContentPath);
+
+        if (targets != null) {
+            for (String target : targets) {
+                int count = removeEntriesFromResolvesMap(target, actualContentPath);
+                if (vanityCounter.longValue() >= count) {
+                    vanityCounter.addAndGet(-count);
```

Comment:
why was this changed from ``-2`` to ``-count``? While it sounds logically, is it a bug in the existing implementation?

Reply:
There was an assumption throughout the code that the # of cache entries per vanity path always is 2. That is an implementation detail, and I tried - hopefully successfully - to return and use the actual number of additions/removals. Note that this (IMHO) only affects that counter, so it probably does not matter at all.